lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -1690,12 +1696,17 @@ Type *UpdateFieldTypeForLegacyLayout(Type *Ty,
 
 StructType *UpdateStructTypeForLegacyLayout(StructType *ST,
                                             DxilTypeSystem &TypeSys,
-                                            Module &M) {
+                                            Module &M,
+                                            bool includeTopLevelResource) {
```

Comment:
`includeTopLevelResource` is used only for the top-level StructuredBuffer resource type case, due to the reliance of that type and type annotation metadata for reflection.  However, if a StructuredBuffer is inside another structure, it should not be included in that context.  This is only for when we are constructing the type annotation for the StructuredBuffer type itself.